azext_iot/tests/iothub/configurations/test_iot_config_int.py

```diff
@@ -800,3 +804,96 @@ def test_device_configurations(self):
+            )
+
+            # Accept IoT Edge module terms offer
+            self.cmd(
+                "iot edge image terms accept --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                checks=offer_checks.append(self.check("accepted", "true"))
+            )
+
+            # Show the accepted IoT Edge module terms offer using URN
+            self.cmd(
+                "iot edge image terms show --urn {}".format(
+                    terms_urn
+                ),
+                checks=offer_checks.append(self.check("accepted", "true"))
+            )
+
+            # Cancel IoT Edge module terms offer
+            self.cmd(
+                "iot edge image terms cancel --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                checks=offer_checks.append(self.check("accepted", "false"))
+            )
+
+            # Error - providing offer, plan and publisher when URN is already provided
+            self.cmd(
+                "iot edge image terms show --urn {} --offer {} --plan {} --publisher {}".format(
+                    terms_urn, terms_offerId, terms_planId, terms_publisherId
+                ),
+                expect_failure=True,
+            )
+
+            # Error - invalid URN format
+            self.cmd(
+                "iot edge image terms show --urn {}".format(
+                    "bad_URN"
+                ),
+                expect_failure=True,
+            )
+
+            # Error - invalid offer
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    "invalid_offer", terms_planId, terms_publisherId
+                ),
+                expect_failure=True,
+            )
+
+            # Error - invalid publisher
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, "invalid_publisher"
+                ),
+                expect_failure=True,
+            )
+        else:
+            # Error - Command run by a service principal user
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                expect_failure=True,
+            )
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup_edge_image_terms_tests(self):
+        # Ensure Edge image offer terms are not accepted before the test starts
+        if self.current_user["type"] == UserTypes.user.value:
+            self.cmd(
+                "iot edge image terms cancel --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                )
+            )
+        yield
```

Comment:
Why yield? There is nothing happening afterwards.

Reply:
Yield signifies that the fixture is the first thing to run when the test class initializes. It's not mandatory but makes the intent clear. The same practice is being followed in central tests.

Reply:
Can you provide some documentation for what you are saying?

In pytest fixtures, upon a yield, the pytest runtime moves on to other fixtures (if any) then executes the test. After the test executes, the pytest runtime will run through fixtures (in reverse order) and execute code after the yield. In other words an effective pattern with yield is a setup before a yield, and a clean up after. https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended

Reply:
Here is an example:

![image](https://user-images.githubusercontent.com/51140335/149390057-eb4639d4-7a98-4abb-9f8b-b449c11391cf.png)

Reply:
Sorry let me clarify, I am asking for evidence on why you think this is a correct usage of yield in a pytest fixture or how a yield of None, followed by no code makes the intent clear.

This example you have provided shows an unneeded usage of yield which we should fix.

My prior comment has a link to the official docs showing usage of yield. Here is a link to the official docs on fixture instantiation order. https://docs.pytest.org/en/6.2.x/fixture.html#fixture-instantiation-order

Reply:
I removed yield from the fixture you have shown and it doesn't make a difference for the Central tests 

![image](https://user-images.githubusercontent.com/28658112/149399923-7c55089f-c161-4dce-94aa-851e457d2066.png)

Reply:
Yeah the 'yield' is not required and the result is same weather or not we put it. I feel it just makes the intent clear to someone reading the code that the statements above yield run as a part of the test setup (before the tests are executed). However, I can remove it if you feel strongly against it?

Reply:
Usage of fixtures in tests imply setups or teardowns for tests to function properly - they provide a fixed baseline for the test to execute reliably.

When I saw the yield followed by nothing, the intent or purpose of it was not clear to me hence this feedback. I shared documentation from pytest explaining proper usage of yield in fixtures, which is not being done here.

yield could be replaced with a return or nothing and the net effect is the same (actually slighter improved perf because pytest does not need to attempt to run code after yield).

I don't understand the benefit.

Reply:
I see your point, will remove it.